src/lua/LuaBody.cpp

```diff
@@ -384,28 +392,26 @@ static int l_body_get_position_rel_to(lua_State *l)
  *
  *   stable
  */
-
 static int l_body_get_altitude_rel_to(lua_State *l)
 {
+	Body *b = LuaObject<Body>::CheckFromLua(1);
 	const Body *other = LuaObject<Body>::CheckFromLua(2);
-	vector3d pos = Pi::player->GetPositionRelTo(other);
-	double center_dist = pos.Length();
-	if (other && other->IsType(ObjectType::TERRAINBODY)) {
-		const TerrainBody *terrain = static_cast<const TerrainBody *>(other);
-		vector3d surface_pos = pos.Normalized();
-		double radius = terrain->GetSystemBody()->GetRadius();
-		if (center_dist <= 3.0 * terrain->GetMaxFeatureRadius()) {
-			radius = terrain->GetTerrainHeight(surface_pos);
+	std::string typeString = LuaPull<const char*>(l, 3, "default");
+	AltitudeType altType = AltitudeType::DEFAULT;
+	if (typeString.compare("default"))
+	{
+		if (auto str = altitudeTypesStringMap.find(typeString); str != altitudeTypesStringMap.end())
+		{
+			altType = str->second;
+		}
+		else
+		{
+			return luaL_error(l, "invalid altitude type: %s", typeString.c_str());
 		}
```

Comment:
Please either use the `EnumStrings` facility to convert a string to an enum value decorated for use with the `scan_enums` script, or (better) use a tri-state boolean (nil, false, true) check here - if `nil`, use AltitudeType::DEFAULT, if `false` use `SEA_LEVEL`, if `true` use `ABOVE_TERRAIN`. There's not really a need to define yet another custom enumeration type for the tiny API call surface at play here, especially when the tri-state semantics map perfectly to the effect of the enumeration.

Reply:
Used boolean approach.

Reply:
As a note for the future - since the enum you previously used here isn't exposed to Lua anymore, it doesn't need to be marked up for use with enum_table/EnumStrings. Having that enum available even though nothing uses it currently is fine, but just a bit messy.